File: task1/task1b/Truck.cpp
#include "Truck.h"
#include <iostream>
#include <string.h>

Truck::Truck(){}

Truck::Truck(int id, const string& type, double speed, double cost): Transport(id, type, speed, cost) {}

void Truck::display_transport_details() {
    Transport::display_transport_details();
    cout << "Motorway List: " << endl;
    for(int i = 0; i < motorwayList.size(); i++){
        cout << motorwayList.at(i) << endl;
    }
}

void Truck::updateMotorwayinList(const string& motorway, int pos){
    motorwayList.at(pos) = motorway;
}


// Setters
void Truck::setMotorway(const string& motorway){
    this->motorway = motorway;
}

void Truck::addToList(const string& motorway){
    motorwayList.push_back(motorway);
}


// Getters
string Truck::getRoute(int pos) const{
    string motorwayAtPos = motorwayList.at(pos);
    return motorwayAtPos;
}

File: task2/task2/src/main/java/com/task2/Truck.java
package com.task2;

import java.util.ArrayList;

public class Truck extends Transport{
    private ArrayList<String> motorwayList = new ArrayList<String>();
    private String motorway;

    public Truck(){

    }

    public Truck(int id, String type, double speed, double cost){
        super(id, type, speed, cost);
    }

    @Override
    public void display_vehicle_details(){
        super.display_vehicle_details();
        System.out.println("Motorway List:");
        for (int i = 0; i < motorwayList.size(); i++){
            System.out.println(motorwayList.get(i));
        }
    }

    // Setters
    public void setMotorway(String motorway){
        this.motorway = motorway;
    }

    public void addToList(String motorway){
        motorwayList.add(motorway);
    }

    public void updateMotorwayinList(String motorway, int pos){
        motorwayList.set(pos, motorway);
    }

    // Getters
    public String getRoute(int pos){
        String motorwayAtPos = motorwayList.get(pos);
        return motorwayAtPos;
    }

}
